src/Domain/Block.cpp

```diff
@@ -19,15 +21,39 @@ Block<VolumeDim>::Block(
     std::unique_ptr<domain::CoordinateMapBase<Frame::Logical, Frame::Inertial,
                                               VolumeDim>>&& stationary_map,
     const size_t id,
-    DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors) noexcept
+    DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors,
+    DirectionMap<VolumeDim, std::unique_ptr<
+                                domain::BoundaryConditions::BoundaryCondition>>
+        external_boundary_conditions) noexcept
     : stationary_map_(std::move(stationary_map)),
       id_(id),
       neighbors_(std::move(neighbors)) {
+  ASSERT(external_boundary_conditions.empty() or
+             external_boundary_conditions.size() ==
+                 2 * VolumeDim - neighbors_.size(),
+         "Must have either no boundary conditions or one boundary condition "
+         "for each external boundary.");
   // Loop over Directions to search which Directions were not set to neighbors_,
   // set these Directions to external_boundaries_.
   for (const auto& direction : Direction<VolumeDim>::all_directions()) {
     if (neighbors_.find(direction) == neighbors_.end()) {
-      external_boundaries_.emplace(std::move(direction));
+      ASSERT(external_boundary_conditions.empty() or
+                 external_boundary_conditions.contains(direction),
+             "Specifying boundary conditions but the external direction "
+                 << direction << " in block " << id_
+                 << " does not have a specified boundary condition.");
+      if (not external_boundary_conditions.contains(direction)) {
```

Comment:
(optional) I find it easier to follow control flow with fewer boolean inversions, so I suggest to swap the `if` and `else` clauses here.